src/software/geom/rectangle.h

```diff
@@ -136,6 +136,17 @@ class Rectangle final
      */
     double distToBoundary(Point p);
 
+    /**
+     * Returns the corner point of the rectangle that is the furthest from the input
+     * point.
+     *
+     * @param p The point to test
+     * @return The corner point that is furthest from the test point
+     */
+    Point furthestCorner(Point p);
+
+    std::vector<Point> corners();
```

Comment:
It would be nice to have a description of which corners are where inside the vector.
operator[](0) is not the most descriptive.